### PR TITLE
ITE: drivers/i2c: remove redundant printk

### PR DESCRIPTION
**Pull request by GTLin08**
ITE: drivers/i2c: remove redundant printk
This printk has been printed in i2c_recovery_bus routine,
so here is redundant.

ITE: drivers/i2c: disable the interrupt
The irq will be enabled at the condition of start or repeat
start of I2C. If timeout occurs without being wake up during
suspend(ex: interrupt is not fired), the irq should be
disabled immediately.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>

---
This change is [Reviewable](https://reviewable.io/reviews/zephyrproject-rtos/zephyr/37099)

### CODE REVIEW

File: drivers/i2c/i2c_ite_it8xxx2.c [hunk 1]
```diff
@@ -805,6 +803,11 @@ static int i2c_it8xxx2_transfer(const struct device *dev, struct i2c_msg *msgs,
 		/* Wait for the transfer to complete */
 		/* TODO: the timeout should be adjustable */
 		res = k_sem_take(&data->device_sync_sem, K_MSEC(100));
+		/*
+		 * Ensure that interrupt can be disabled immediately when
+		 * the timeout occurs.
+		 */
+		irq_disable(config->i2c_irq_base);
```
Comment (brockus-zephyr) [inline drivers/i2c/i2c_ite_it8xxx2.c:810]: When does this get enabled/re-enabled?

Comment (sjg20) [reply drivers/i2c/i2c_ite_it8xxx2.c:810]: Yes the commit messages needs a motivation. It should always be the first thing in the commit message

**Review by sjg20 — CHANGES_REQUESTED** [review]

File: drivers/i2c/i2c_ite_it8xxx2.c (hunk 1 above)
Comment (GTLin08) [reply drivers/i2c/i2c_ite_it8xxx2.c:812]: The commit message and  comment are modified.

**Review by brockus-zephyr — APPROVED** [review]
> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/zephyrproject-rtos/zephyr/37099)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @jackrosenthal and @keith-zephyr)

**Review by jackrosenthal — APPROVED** [review]

**Review by sjg20 — APPROVED** [review]

**Pull request merged**